packages/core/services/geocoder-service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import { bindCallback, from, Observable, of, throwError } from 'rxjs';
+import { flatMap, switchMap } from 'rxjs/operators';
+import { Geocoder, GeocoderRequest, GeocoderResult, GeocoderStatus } from './google-maps-types';
+import { MapsAPILoader } from './maps-api-loader/maps-api-loader';
+
+declare var google: any;
+
+@Injectable()
+export class AgmGeocoder {
+  protected readonly geocoder$: Observable<Geocoder>;
+
+  constructor(loader: MapsAPILoader) {
+    this.geocoder$ = from(loader.load()).pipe(
+      flatMap(() => this._createGeocoder())
+    );
+  }
+
+  getGeocoder$() {
```

Comment:
why do you want to give public access to the underlying geocoder?

Reply:
I am not saying it's wrong, I just want to hear your reasoning

Reply:
To be fair, I've done it because I tried to follow the structure of fit-bounds.ts service, but I can see that as of now there's no reason to have access to this.